docs: Move memory tuning to an advanced topic

gavinelder opened the pull request:
Follow-up to #891 which moves the JVM memory tuning configuration to a dedicated page under Advanced topics.

Manually configuring JVM settings can have adverse consequences and should only be done based on observed performance issues. Specifying JVM parameters on deployments by default can negatively impact customers and we should rely on the frameworks default memory management systems. 

I am following up with a detailed metrics observation guidance to help give customers greater insight into their application performance.

gavinelder committed:
docs: Move memory tuning to an advanced topic

gavinelder committed:
Merge branch 'master' into ge/fix/jvm-memory-tuning

justinegeffen committed:
Update jvm-memory-tuning.md

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

justinegeffen committed:
Update JVM memory tuning documentation metadata

Added creation date and tags to the JVM memory tuning documentation.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

justinegeffen committed:
Update JVM memory tuning documentation metadata

Add creation date and tags for JVM memory tuning documentation

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

justinegeffen committed:
Update JVM memory tuning doc with metadata

Add creation date and tags to JVM memory tuning documentation

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

justinegeffen committed:
Update JVM memory tuning doc with date and tags

Added creation date and tags to JVM memory tuning documentation.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

justinegeffen committed:
Update JVM memory tuning documentation metadata

Added creation date and tags to JVM memory tuning documentation.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

gavinelder committed:
Merge branch 'master' into ge/fix/jvm-memory-tuning

gwright99 commented on platform-enterprise_versioned_sidebars/version-25.2-sidebars.json:
```diff
@@ -60,7 +60,8 @@
               "enterprise/advanced-topics/custom-launch-container",
               "enterprise/advanced-topics/firewall-configuration",
               "enterprise/advanced-topics/seqera-container-images",
-              "enterprise/advanced-topics/content-security-policy"
```
Is there a reason the CSP link only starts in v25.2 docs?

gwright99 commented on platform-enterprise_versioned_docs/version-24.2/enterprise/advanced-topics/jvm-memory-tuning.md:
```diff
@@ -0,0 +1,66 @@
+---
```
Rather than have multiple copies of the same text in various versions, is it possible to make these pages DRY and link back to `platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md`?

gwright99 commented on platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md:
```diff
@@ -0,0 +1,66 @@
+**Increase heap memory (`-Xmx`)** if you see:
+
+- `OutOfMemoryError: Java heap space` errors in logs
+- Garbage collection pauses affecting performance
```
How are we expecting these metrics to be visible. IIRC you dont get memory metrics on the standards EC2 monitoring package. Do we expect the client to upgrade their monitoring system / be using an aggregating agent like Datadog?

gwright99 commented on platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md:
```diff
@@ -0,0 +1,66 @@
+
+**Increase direct memory (`MaxDirectMemorySize`)** if you see:
+
+- `OutOfMemoryError: Direct buffer memory` errors in logs
```
Increase relative to what?
- Grant more memory at the expense of heap?
- Grant more memory at the expense of overhead?
- Something else?

gwright99 reviewed (changes requested):
Comments and first thoughts within. Happy to discuss further async.

gavinelder commented on platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md:
```diff
@@ -0,0 +1,66 @@
+
+## Memory parameters
+
+Set JVM memory parameters using the `JAVA_OPTS` environment variable. The following parameters control memory allocation:
```
Something like 

JVM memory parameters can be configured using the `JAVA_OPTS` environment variable. The following parameters control memory allocation:

gavinelder commented on platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md:
```diff
@@ -0,0 +1,66 @@
+
+Set JVM memory parameters using the `JAVA_OPTS` environment variable. The following parameters control memory allocation:
+
+| Parameter                  | Description                                                                                                                         |
```
I could not find an easily accessible external source which did not go deep into compiler and runtime behaviors

gavinelder commented on platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md:
```diff
@@ -0,0 +1,66 @@
+
+- `OutOfMemoryError: Direct buffer memory` errors in logs
+- High concurrent workflow launch rates (more than 100 simultaneous workflows)
+- Large configuration payloads or extensive API usage
```
Happy to drop these

gavinelder committed:
feat: add platform monitoring guide (#954)

* feat: add platform monitoring guide

* feat: Add obs signals

* Update monitoring.md

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Update monitoring.md

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Clean up blank lines in monitoring.md

Removed unnecessary blank lines in the monitoring documentation.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Update monitoring.md

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Refine Seqera Platform monitoring documentation

Updated monitoring documentation for Seqera Platform, including corrections to metric names and descriptions.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Revise monitoring.md for terminology updates

Updated monitoring documentation for Seqera Platform to reflect changes in terminology and improve clarity.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Refine Seqera Platform Monitoring documentation

Updated monitoring documentation to improve clarity and consistency in terminology.

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Apply suggestion from @justinegeffen

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Apply suggestion from @justinegeffen

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Apply suggestion from @justinegeffen

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Apply suggestion from @justinegeffen

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

* Apply suggestion from @justinegeffen

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>

---------

Signed-off-by: Justine Geffen <justinegeffen@users.noreply.github.com>
Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

justinegeffen commented:
Tracking the intent of this PR in **EDU-1480** — moving JVM memory tuning to Advanced topics and reframing it as diagnostic-led rather than an install step. The follow-on metrics-observation guidance you mentioned is noted there too, against the monitoring guide from #954 / #1147. Branch `ge/fix/jvm-memory-tuning` is still on the remote.
